### PR TITLE
hiop: add version v0.5.4

### PR DESCRIPTION
**Pull request by ashermancinelli**
CC @CameronRutherford

### CODE REVIEW

**Review by cameronrutherford — DISMISSED** [review]

File: var/spack/repos/builtin/packages/hiop/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/hiop/package.py]: I see that maintainer approved, but I wonder if we should have a tagged version depend on `magma@master` (which I assume is a moving target). @adamjstewart Do you have opinions on this?

**Review by alalazo — COMMENTED** [review]
Recipe LGTM, one concern on `magma@master` being a dependency of a tagged version.

File: var/spack/repos/builtin/packages/hiop/package.py
Comment (ashermancinelli) [reply var/spack/repos/builtin/packages/hiop/package.py]: This is very much a concern for us as well, but Magma 2.6.2 was supposed to be released last November and we've needed this release for some time now. Another option might be to add a deprecated version to magma that points to the commit I mention in the comment just for this hiop (depreciated so other users won't accidentally use it).

Comment (alalazo) [reply var/spack/repos/builtin/packages/hiop/package.py]: Would it be fine to:
1. Mark `magma@2.6.1` as preferred
2. Add a placeholder for the new magma release, e.g. `2.6.2rc1`, and pin it to that commit (without saying it's deprecated, which will make Spack trying actively to avoid it)

? In this way `hiop` will depend on a fixed commit rather than a moving branch.

Comment (ashermancinelli) [reply var/spack/repos/builtin/packages/hiop/package.py]: That seems like a great solution! With the sign off of Magma folks I'm happy to make this change. Does it belong in this PR?

Comment (alalazo) [reply var/spack/repos/builtin/packages/hiop/package.py]: > Does it belong in this PR?

Yes, I would do it in this PR since it is required to have the new version of `hiop`.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/hiop/package.py]: Yes, I would add a 2.6.2rc1 version to magma, make 2.6.1 preferred, and have hiop require `magma@2.6.2:`.

File: var/spack/repos/builtin/packages/hiop/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hiop/package.py]:
```suggestion
        ('2.6.2', '0.5.4'),
```
Spack actually sorts 2.6.2rc1 as being newer than 2.6.2, so if you want to be able to use 2.6.2 when it's fully released you'll need to change this

Comment (ashermancinelli) [reply var/spack/repos/builtin/packages/hiop/package.py]: Hm, that's in interesting way for spack to sort versions 😅 . What do you think about `2.6.1.99` so it doesn't conflict with 2.6.2 when it's released? (CCing magma maintainers, seems like spackbot doesn't pick them up if the package is added later on @G-Ragghianti @luszczek @stomov)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/hiop/package.py]: > so it doesn't conflict with 2.6.2 when it's released?

What do you mean?

Comment (ashermancinelli) [reply var/spack/repos/builtin/packages/hiop/package.py]: Oh, I see. depending on `2.6.2:` is better than `2.6.2rc1:`. Thanks!

**Review by adamjstewart — APPROVED** [review]
LGTM, will give the magma maintainers more time to review. Ping me in a week if you never hear back from anyone else.

File: var/spack/repos/builtin/packages/hiop/package.py
Comment (stomov) [reply var/spack/repos/builtin/packages/hiop/package.py]: Related to magma, we continued adding functionalities and didn't release 2.6.2 as planned. We can mark current master as 2.6.2 by the end of this week to resolve problems like this.

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/hiop/package.py
Comment (cameronrutherford) [inline var/spack/repos/builtin/packages/hiop/package.py:77]: I'm conflicted about this. This is very clean, but I wish there was syntax along the lines of the following:

```python
depends_on('magma@version:', when='@version:^magma')
```

Realistically this is a small gripe, but it is frustrating having to re-specify dependency logic when attempting to define version constraints.

**Review by cameronrutherford — APPROVED** [review]

File: var/spack/repos/builtin/packages/magma/package.py
Comment (G-Ragghianti) [inline var/spack/repos/builtin/packages/magma/package.py:26]: If this causes a ordering problem with the release of 2.6.2, then we can either remove the rc1 version or mark the 2.6.2 as preferred.  Looks fine.

**Review by G-Ragghianti — APPROVED** [review]

File: var/spack/repos/builtin/packages/magma/package.py
Comment (ashermancinelli) [reply var/spack/repos/builtin/packages/magma/package.py:26]: I'm happy to remove 2.6.2rc1 once 2.6.2 proper is released.

**Review by stomov — APPROVED** [review]

File: var/spack/repos/builtin/packages/hiop/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/hiop/package.py:77]: See #14002

**Pull request merged**
